wgsl/index.bs

```diff
@@ -17795,40 +17795,184 @@ fn atomicLoad(atomic_ptr: ptr<AS, atomic<T>, read_write>) -> T
+</div>
+
+#### `atomicAnd` #### {#atomic-and}
+
+```wgsl
 fn atomicAnd(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
+```
+
+Atomically performs a bitwise AND operation on the atomic object pointed to by `atomic_ptr`
+with the value `v`, and returns the original value stored in the atomic object before the operation.
+
+<div class='example wgsl global-scope' heading='Operation of `atomicAnd` as a function'>
+  <xmp highlight=wgsl>
+// All operation performed atomically
+fn atomicAnd(atomic_ptr: ptr<AS, atomic<T>, read_write>, v : T) -> T {
+  let old = *atomic_ptr;
+  *atomic_ptr = old & v;
+  return old;
+}
+  </xmp>
+</div>
+
+#### `atomicOr` #### {#atomic-or}
+
+```wgsl
 fn atomicOr(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
+```
+
+Atomically performs a bitwise OR operation on the atomic object pointed to by `atomic_ptr`
+with the value `v`, and returns the original value stored in the atomic object before the operation.
+
+<div class='example wgsl global-scope' heading='Operation of `atomicOr` as a function'>
+  <xmp highlight=wgsl>
+// All operation performed atomically
+fn atomicOr(atomic_ptr: ptr<AS, atomic<T>, read_write>, v : T) -> T {
+  let old = *atomic_ptr;
+  *atomic_ptr = old | v;
+  return old;
+}
+  </xmp>
+</div>
+
+#### `atomicXor` #### {#atomic-xor}
+
+```wgsl
 fn atomicXor(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
 ```
-Each function performs the following steps atomically:
 
-1. Load the original value pointed to by `atomic_ptr`.
-2. Obtains a new value by performing the operation (e.g. max) from the function
-    name with the value |v|.
-3. Store the new value using `atomic_ptr`.
+Atomically performs a bitwise XOR operation on the atomic object pointed to by `atomic_ptr`
+with the value `v`, and returns the original value stored in the atomic object before the operation.
+
+<div class='example wgsl global-scope' heading='Operation of `atomicXor` as a function'>
+  <xmp highlight=wgsl>
+// All operation performed atomically
+fn atomicXor(atomic_ptr: ptr<AS, atomic<T>, read_write>, v : T) -> T {
+  let old = *atomic_ptr;
+  *atomic_ptr = old ^ v;
+  return old;
+}
+  </xmp>
+</div>
 
-Each function returns the original value stored in the atomic object.
+### `atomicExchange` ### {#atomic-exchange}
 
 ```wgsl
 fn atomicExchange(atomic_ptr: ptr<AS, atomic<T>, read_write>, v: T) -> T
 ```
 
-Atomically stores the value `v` in the atomic object pointed to
-`atomic_ptr` and returns the original value stored in the atomic object.
+Atomically stores the value `v` in the atomic object pointed to by
+`atomic_ptr` and returns the original value stored in the atomic object before the operation.
+
+<div class='example wgsl global-scope' heading='Operation of `atomicExchange` as a function'>
+  <xmp highlight=wgsl>
+// All operation performed atomically
+fn atomicExchange(atomic_ptr: ptr<AS, atomic<T>, read_write>, v : T) -> T {
+  let old = *atomic_ptr;
+  *atomic_ptr = v;
+  return old;
+}
+  </xmp>
```

Comment:
This is missing a </div>. Probably my fault.

Reply:
Fixed, TYVM!